cmd/lit-af/shell.go

```diff
@@ -229,141 +235,210 @@ func (lc *litAfClient) Ls(textArgs []string) error {
-		for _, peer := range pReply.Connections {
-			fmt.Fprintf(color.Output, "%s %s (%s)\n",
-				lnutil.White(peer.PeerNumber), peer.RemoteHost, peer.LitAdr)
+
+	if cmd == "conns" || all {
+		err := lc.Call("LitRPC.ListConnections", nil, pReply)
+		if err != nil {
+			return err
 		}
-	}
 
-	err = lc.Call("LitRPC.ChannelList", nil, cReply)
-	if err != nil {
-		return err
-	}
-	if len(cReply.Channels) > 0 {
-		fmt.Fprintf(color.Output, "\t%s\n", lnutil.Header("Channels:"))
+		if all {
+			fmt.Fprintf(color.Output, "\t%s\n", lnutil.Header("Peers:"))
+		}
+
+		if len(pReply.Connections) > 0 {
+			for _, peer := range pReply.Connections {
+				fmt.Fprintf(color.Output, "%s %s (%s)\n",
+					lnutil.White(peer.PeerNumber), peer.RemoteHost, peer.LitAdr)
+			}
+		} else {
+			fmt.Printf("none\n")
+		}
 	}
 
-	sort.Slice(cReply.Channels, func(i, j int) bool {
-		return cReply.Channels[i].Height < cReply.Channels[j].Height
-	})
+	if cmd == "chans" || all {
+		err := lc.Call("LitRPC.ChannelList", nil, cReply)
+		if err != nil {
+			return err
+		}
+
+		if all {
+			fmt.Fprintf(color.Output, "\t%s\n", lnutil.Header("Channels:"))
+		}
+
+		if len(cReply.Channels) > 0 {
+
+			sort.Slice(cReply.Channels, func(i, j int) bool {
+				return cReply.Channels[i].Height < cReply.Channels[j].Height
+			})
+
+			var closedChannels []litrpc.ChannelInfo
+			var openChannels []litrpc.ChannelInfo
+			for _, c := range cReply.Channels {
+				if c.Closed {
+					closedChannels = append(closedChannels, c)
+				} else {
+					openChannels = append(openChannels, c)
+				}
+			}
 
-	var closedChannels []litrpc.ChannelInfo
-	var openChannels []litrpc.ChannelInfo
-	for _, c := range cReply.Channels {
-		if c.Closed {
-			closedChannels = append(closedChannels, c)
+			for _, c := range closedChannels {
+				fmt.Fprintf(color.Output, lnutil.Red("Closed:       \n"))
+				fmt.Fprintf(
+					color.Output,
+					"%s (peer %d) type %d %s\n\t cap: %s bal: %s h: %d state: %d data: %x pkh: %x\n",
+					lnutil.White(c.CIdx), c.PeerIdx, c.CoinType,
+					lnutil.OutPoint(c.OutPoint),
+					lnutil.SatoshiColor(c.Capacity), lnutil.SatoshiColor(c.MyBalance),
+					c.Height, c.StateNum, c.Data, c.Pkh)
+			}
+
+			for _, c := range openChannels {
+				if c.Height == -1 {
+					c := color.New(color.FgGreen).Add(color.Underline)
+					c.Printf("Unconfirmed:")
+					fmt.Fprintf(color.Output, lnutil.Green("  "))
+					//needed for preventing the underline from extending
+				} else {
+					fmt.Fprintf(color.Output, lnutil.Green("Open:         \n"))
+				}
+				fmt.Fprintf(
+					color.Output,
+					"%s (peer %d) type %d %s\n\t cap: %s bal: %s h: %d state: %d data: %x pkh: %x\n",
+					lnutil.White(c.CIdx), c.PeerIdx, c.CoinType,
+					lnutil.OutPoint(c.OutPoint),
+					lnutil.SatoshiColor(c.Capacity), lnutil.SatoshiColor(c.MyBalance),
+					c.Height, c.StateNum, c.Data, c.Pkh)
+			}
 		} else {
-			openChannels = append(openChannels, c)
+			fmt.Printf("none\n")
```

Comment:
similar to the peers comment, I think we're better off not printing channels if we don't have any like we do now. If this is too difficult, we could keep this though :)